plugins/destination/elasticsearch/client/test_connection.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/rs/zerolog"
+)
+
+const (
+	codeInvalidSpec      = "INVALID_SPEC"
+	codeConnectionFailed = "CONNECTION_FAILED"
+	codeUnauthorized     = "UNAUTHORIZED"
+	codeUnreachable      = "UNREACHABLE"
+)
+
+const (
+	errUnauthorized = "an error happened during the Info query execution: EOF"
+	errUnreachable  = "an error happened during the Info query execution: dial tcp"
```

Comment:
I think we should check for the `net.OpErr` in the error chain instead of looking at the message

Reply:
Changed the approach here as it was pretty nasty, btw you were right about the net.OpErr